kde-misc/kdialog/kdialog-22.12.2.recipe

```diff
@@ -0,0 +1,118 @@
+SUMMARY="Display dialog boxes from shell scripts"
+DESCRIPTION="kdialog allows you to display dialog boxes from shell scripts.
+The syntax is very much inspired from the 'dialog' command
+(which shows text mode dialogs)."
+HOMEPAGE="https://invent.kde.org/utilities/kdialog/"
+COPYRIGHT="2006-2023 KDialog Team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://invent.kde.org/utilities/kdialog/-/archive/v22.12.2/kdialog-v22.12.2.tar.gz"
+CHECKSUM_SHA256="35ad54955a48def6a13b726da17cf9b95119b794dd9ec183a403c197d25fc4de"
+SOURCE_DIR="kdialog-v$portVersion"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
```

Comment:
This needs to be a code block that changes the value depending on the architecture.  For example the [coreutils recipe](https://github.com/haikuports/haikuports/blob/b6ae6333c0baad9e943b4e482fb97925e92894c0/sys-apps/coreutils/coreutils-9.1.recipe#L29).

Reply:
I'm confused, why would I change `binDir` on ` x86_gcc2 ` if ` x86_gcc2` is a disabled architecture?

Reply:
By default, the binaries built for the x86 secondary architecture are installed in bin/x86 to avoid conflicts with binaries from the same recipe built with gcc2.

When gcc2 is not used, this is not useful, so the default must be overriden to put the binaries in bin/ instead

Reply:
Should now be done

Reply:
It needs the full block including the `$commandSuffix` part which will be used later in the recipe.

Reply:
ok, `$commandSuffix` is now added on `gcc2`.
I also added a patch to fix an issue with ``kdialog --getexistingdirectory`` that didn't allow to actually select a directory on Haiku